pkg/storer/internal/reserve/reserve.go

```diff
@@ -24,11 +25,27 @@ import (
 	"github.com/ethersphere/bee/v2/pkg/swarm"
 	"github.com/ethersphere/bee/v2/pkg/topology"
 	"golang.org/x/sync/errgroup"
-	"resenje.org/multex"
 )
 
 const reserveScope = "reserve"
 
+type multexLock struct {
```

Comment:
Definitely this is not a good approach for the multexLock could lead to deadlocks when trying to lock multiple items. Why the change from resenje multex? cc. @janos